setup-dev-env.sh

```diff
@@ -179,7 +180,7 @@ fi
 # Install pipx for ansible
 if ! (python3 -m pipx --version >/dev/null 2>&1); then
     sudo apt-get -y update
-    python3 -m pip install --user pipx
+    python3 -m pip install --user pipx --break-system-packages
```

Comment:
adding --break-system-packages would cause error in Ubuntu22.04 with the following error message:
```
no such option: --break-system-packages
```

Instead, we can set PIP_BREAK_SYSTEM_PACKAGES=1.
Same applies to all the other places.

```suggestion
    PIP_BREAK_SYSTEM_PACKAGES=1 python3 -m pip install --user pipx
```

Reply:
https://pipx.pypa.io/stable/installation/

> On Linux:
> Ubuntu 23.04 or above
> sudo apt update
> sudo apt install pipx
> pipx ensurepath

Breaking system packages is a very bad idea, we should avoid this by either utilizing apt packages or a virtualized python env like miniconda or venv.

Reply:
Okay, then I think we can use apt install
```suggestion
	sudo apt install pipx
```

Reply:
```suggestion
    UBUNTU_VERSION=$(lsb_release -rs)
    echo "Detected Ubuntu version: $UBUNTU_VERSION"

    if [[ "$UBUNTU_VERSION" == "22.04" ]]; then
        echo "Installing pipx using pip (for Ubuntu 22.04)..."
        python3 -m pip install --user pipx
    else
        echo "Installing pipx using apt (for Ubuntu > 22.04)..."
        sudo apt update
        sudo apt install -y pipx
    fi
```

To keep compatibility with older version.

Reply:
Do we need to keep the old one? We should be able to install pipx with apt for Ubuntu22.04 as well.

Reply:
I have double check that ubuntu 22.04 support apt install pipx,
I will update this file to use `sudo apt install pipx` for both ubuntu version